modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/formik-fields/MultiSelect/index.jsx

```diff
@@ -0,0 +1,130 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayMultiSelect from '@clayui/multi-select';
+import ClaySticker from '@clayui/sticker';
+import classNames from 'classnames';
+import {useField, useFormikContext} from 'formik';
+import {useEffect, useState} from 'react';
+import {Badge} from '../..';
+import {required, validateEmailsArray} from '../../../utils/validations.form';
+
+const MultiSelect = ({
+	groupStyle,
+	helper,
+	items,
+	label,
+	metaErrorCallback,
+	onChanges,
+	sourceItems,
+	validations,
+	values,
+	...props
+}) => {
+	const [emailsAvailable] = useState(sourceItems);
+
+	const formik = useFormikContext();
+	if (props.required) {
+		validations = validations
+			? [...validations, () => required(values.length)]
+			: [() => required(values.length)];
+	}
+	const handleChange = (item) => {
+		return onChanges(item);
+	};
+	const validateMultiSelect = () => {
+		const unfilledField = validations
+			.map((validation) => validation(values))
+			.filter((error) => !!error);
+
+		const emailErrors = validateEmailsArray(
+			values.map((item) => item?.email || item?.label),
+			emailsAvailable
+		);
+
+		return unfilledField.length ? unfilledField[0] : emailErrors;
+	};
+
+	const [field, meta] = useField({
+		...props,
+		validate: validateMultiSelect,
+	});
+
+	useEffect(() => {
+		formik.setFieldValue(props.name, values);
+		formik.validateField(props.name);
+		metaErrorCallback(!meta.error);
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [values]);
+
+	return (
+		<div className="multi-select-container">
+			<ClayForm.Group
+				className={classNames('w-100', {
+					groupStyle,
+					'has-error': meta.touched && meta.error,
+					'has-success': meta.touched && !meta.error,
+				})}
+			>
+				<label>
+					{`${label} `}
+
+					{props.required && (
+						<span className="inline-item-after reference-mark text-warning">
+							<ClayIcon symbol="asterisk" />
+						</span>
+					)}
+				</label>
+
+				<ClayMultiSelect
+					{...field}
+					{...props}
+					items={items}
+					onChange={() => handleChange(window.event?.target?.value)}
```

Comment:
Never get the event from window, bad practice, use the onChange event instead

```suggestion
					onChange={(event) => onChange(event?.target?.value)}
```